airflow/migrations/versions/2c6edca13270_resource_based_permissions.py

```diff
@@ -161,7 +161,7 @@
         (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_INSTANCE),
     ],
     ("Airflow", "can_trigger"): [
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG),
```

Comment:
Changing the migration won't affect anyone who has already upgrade to 2.0.0 - do we need a new migration to fix this up?

Reply:
This will will help anyone who goes straight from 1.X to 2.0.1, which is probably most of the users.

We shouldn't need a migration, since triggering also requires `Dag Run.can_create`, which is only included in `>= User`. `Dag.can_edit` is also included in `>= User`, so either way, someone with default roles won't be able to trigger dags unless they have `User` or higher. 

This might change the behavior of users with custom roles. If someone has `Viewer` + a custom role with `Dag Run.can_create`, or they manually added `Dag Run.can_create` to `Viewer`, then they will lose access until they add `Dag.can_edit` to the custom role. I suspect this is a small group of users.